### PR TITLE
[SYCL] Disable vectorization loop pipeline 

### PR DESCRIPTION
**Pull request by MrSidims**
Vectorization is target dependent optimization, hence we shall
disable it for SPIR target.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Review by bader — DISMISSED** [review]
Please, fix CI regressions.

File: sycl/test/basic_tests/stream/stream.cpp [hunk 1]
```diff
@@ -1,7 +1,3 @@
-// TODO: Enable compilation w/o -fno-sycl-early-optimizations option.
```
Comment (vladimirlaz) [inline sycl/test/basic_tests/stream/stream.cpp:1]: Looks like tests are still failing

Comment (MrSidims) [reply sycl/test/basic_tests/stream/stream.cpp:1]: Yeah, it was passing with a previous patch set. Converting PR back to draft for now.

**Comment by vladimirlaz** [comment]
LGTM but there is no RT part anymore so my approval is not needed.

**Comment by bader** [comment]
I think (part of) these changes are not needed with this patch and should be reverted: https://github.com/intel/llvm/commit/20921b10e722e87b4d83665b8bf6e525c932a0ea. Right?

**Pull request closed without merging**